### PR TITLE
Support for disabling/enabling URL previews per-user, per-room and per-user-per-room

### PR DESCRIPTION
**Pull request by ara4n**

### CODE REVIEW

File: src/UserSettingsStore.js [hunk 1]
```diff
@@ -113,6 +113,35 @@ module.exports = {
...
+
+    getSyncedSetting: function(type) {
+        var settings = this.getSyncedSettings();
+        return settings[type];
+    },
+
+    setSyncedSetting: function(type, value) {
+        var settings = this.getSyncedSettings();
+        settings[type] = value;
+        // FIXME: handle errors
```
Comment (dbkr) [inline src/UserSettingsStore.js]: For this function, would this just be a question of returning the promise?

File: src/component-index.js [hunk 2]
```diff
@@ -74,6 +74,8 @@ module.exports.components['views.messages.TextualEvent'] = require('./components
 module.exports.components['views.messages.UnknownBody'] = require('./components/views/messages/UnknownBody');
 module.exports.components['views.room_settings.AliasSettings'] = require('./components/views/room_settings/AliasSettings');
 module.exports.components['views.room_settings.ColorSettings'] = require('./components/views/room_settings/ColorSettings');
+module.exports.components['views.room_settings.UrlPreviewSettings'] = require('./components/views/room_settings/UrlPreviewSettings');
+module.exports.components['views.rooms.Autocomplete'] = require('./components/views/rooms/Autocomplete');
```
Comment (dbkr) [inline src/component-index.js]: Is this autocomplete one supposed to be here?

File: src/components/structures/UserSettings.js [hunk 3]
```diff
@@ -261,6 +261,63 @@ module.exports = React.createClass({
...
+            {
+                id: 'useCompactLayout',
+                label: 'Use compact timeline layout',
+            },
+            {
+                id: 'useFixedWidthFont',
+                label: 'Use fixed width font',
+            },
+        */
+        ];
```
Comment (dbkr) [inline src/components/structures/UserSettings.js]: Presumably these things will be coming soon?

**Comment by dbkr** [comment]
OK - the spurious autocomplete looks important. I assume the synced settings stuff got rolled in as it's also in the pipeline, probably shouldn't be part of this as it's not related, but probably too much effort to pick them apart if it will follow soon.

File: src/UserSettingsStore.js (hunk 1 above)
Comment (ara4n) [inline src/UserSettingsStore.js]: hm, howso? it's called from 

```
onChange={ e => UserSettingsStore.setSyncedSetting(setting.id, e.target.checked) }
```

so surely the promise won't go anywhere sensible?

Have made them return anyway, but the FIXME is still there....

File: src/components/structures/UserSettings.js (hunk 3 above)
Comment (ara4n) [inline src/components/structures/UserSettings.js]: yes, it's just laying the groundwork for now.

File: src/component-index.js (hunk 2 above)
Comment (ara4n) [inline src/component-index.js]: it was added by reskindex.  the new autocomplete component exists though and is checked into git by aviral, so why shouldn't it be here?

**Comment by ara4n** [comment]
@dbkr: ptal

File: src/UserSettingsStore.js (hunk 1 above)
Comment (dbkr) [inline src/UserSettingsStore.js]: Oh, indeed - I just meant then at least this function is doing "the right thing", even if the error handling isn't wired up higher up.

File: src/component-index.js (hunk 2 above)
Comment (dbkr) [inline src/component-index.js]: Ah, fair enough.

**Pull request merged**